src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -547,7 +547,7 @@ public void visitToken(DetailAST ast) {
             final boolean isStatic = ast.getType() == TokenTypes.STATIC_IMPORT;
             importToGroupList.add(new ImportDetails(importFullPath,
                     lineNo, endLineNo, getImportGroup(isStatic, importFullPath),
-                    isStatic));
+                    isStatic, ast));
```

Comment:
@developerhb We no longer need the `startLineNumber` and `endLineNumber` fields in `ImportDetails`. Please replace their getters with `ast.getLineNo()` and `ast.getLastChild().getLineNo()`.

Reply:
@pbludov 

Should I perform the following change?

```java
public int getStartLineNumber() {
            return startLineNumber;
        }
```

to

```java
public int getStartLineNumber() {
            return importAST.getLineNo();
        }
```

Reply:
> Should I perform the following change?

Exactly.

Reply:
@pbludov Do I need to re-generate diff report after this change?

Reply:
> @pbludov Do I need to re-generate diff report after this change?

Yes. You need to rebase on latest master and regenerate diff report.

Reply:
@pbludov 
Okay i will make the changes and create new diff report.

Reply:
@pbludov
 I think I should also remove the attributes `startLineNumber` and `endLineNumber`, because I am getting `unread field` exception in `mvn clean verify`. This would also lead to change in the constructor call. Should I go for this?

Reply:
>  Should I go for this?

Yes, please.

Reply:
Done